radio/sdcard/horus/WIDGETS/BattCheck/main.lua

```diff
@@ -177,40 +195,46 @@ end
 
 --- Size is 160x30 1/8th
 local function zoneSmall(zone)
-  local myBatt = {["x"]=0, ["y"]=0, ["w"]=75, ["h"]=28, ["segments_w"]=15, ["color"]=WHITE, ["cath_w"]=6, ["cath_h"]=20}
+  local myBatt = {["x"]=0, ["y"]=0, ["w"]=155, ["h"]=35, ["segments_w"]=25, ["color"]=WHITE, ["cath_w"]=6, ["cath_h"]=20}
   local mySensor = getCels(zone.options.Sensor)
+  local isHaveData = (type(mySensor) == "table")
 
-  lcd.setColor(CUSTOM_COLOR, zone.options.Color)
-  if type(mySensor) == "table" then
-    local myString = string.format("%2.1fV", getCellSum(mySensor)).." ("..getCellCount(mySensor).."S)"
-    local percent = getCellPercent(getCellAvg(mySensor))
-    lcd.drawText(zone.zone.x + zone.zone.w, zone.zone.y + 22, myString, RIGHT + SMLSIZE + CUSTOM_COLOR + shadowed)
-    lcd.drawText(zone.zone.x + zone.zone.w, zone.zone.y, percent.."%", RIGHT + MIDSIZE + CUSTOM_COLOR + shadowed)
-    -- fils batt
-    lcd.setColor(CUSTOM_COLOR, getPercentColor(percent))
-    lcd.drawGauge(zone.zone.x+2, zone.zone.y+2, 75, zone.zone.h - 4, percent, 100, CUSTOM_COLOR)
-    -- draws bat
-    lcd.setColor(CUSTOM_COLOR, WHITE)
-    lcd.drawRectangle(zone.zone.x + myBatt.x, zone.zone.y + myBatt.y, myBatt.w, myBatt.h, CUSTOM_COLOR, 2)
-    lcd.drawFilledRectangle(zone.zone.x + myBatt.x + myBatt.w, zone.zone.y + myBatt.h/2 - myBatt.cath_h/2, myBatt.cath_w, myBatt.cath_h, CUSTOM_COLOR)
-    for i=1, myBatt.w - myBatt.segments_w, myBatt.segments_w do
-      lcd.drawRectangle(zone.zone.x + myBatt.x + i, zone.zone.y + myBatt.y, myBatt.segments_w, myBatt.h, CUSTOM_COLOR, 1)
-    end
+  local percent = getCellPercent(getCellAvg(mySensor))
+
+  -- draws bat
+  lcd.setColor(CUSTOM_COLOR, WHITE)
+  lcd.drawRectangle(zone.zone.x + myBatt.x, zone.zone.y + myBatt.y, myBatt.w, myBatt.h, CUSTOM_COLOR, 2)
+  lcd.drawFilledRectangle(zone.zone.x + myBatt.x + myBatt.w, zone.zone.y + myBatt.h/2 - myBatt.cath_h/2, myBatt.cath_w, myBatt.cath_h, CUSTOM_COLOR)
+
+  local mainValue =  getCellMin(mySensor)
+  local secondaryValue = getCellSum(mySensor)
+
+  -- fils batt
+  lcd.setColor(CUSTOM_COLOR, getPercentColor(percent))
+  lcd.drawGauge(zone.zone.x+2, zone.zone.y+2, myBatt.w-4, zone.zone.h, percent, 100, CUSTOM_COLOR)
+  -- write text
+  if isHaveData then
+    lcd.setColor(CUSTOM_COLOR, zone.options.Color)
+    local topLine = string.format("%2.1fV      %2.0f%%", mainValue, percent)
+    lcd.drawText(zone.zone.x + 20, zone.zone.y +2, topLine , MIDSIZE + CUSTOM_COLOR + shadowed)
   else
-    lcd.drawText(zone.zone.x, zone.zone.y+10, "No FLVSS sensor data", LEFT + SMLSIZE + INVERS + CUSTOM_COLOR + shadowed)
+    lcd.drawText(zone.zone.x + 10, zone.zone.y+10, "No Voltage Data", LEFT + SMLSIZE + INVERS + CUSTOM_COLOR + shadowed)
```

Comment:
If you point the widget to RxBt, there is voltage info, just not the right type, maybe no cells data is better ?

Reply:
good comment, will be changed